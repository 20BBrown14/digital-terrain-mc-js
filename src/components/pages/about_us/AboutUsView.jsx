import React from 'react';
import { List, Card, Typography } from 'antd';
import './AboutUs.css';
import adminModsUUID from '../../../constants/adminModsUUID.json';
import veterans from '../../../constants/veterans.json';

const buildHeadDisplayComponents = (UUIDInformation) => (
  Object.keys(UUIDInformation).map((playerName) => (
    {
      image: (
        <img
          className="about-us-card-cover"
          src={`https://crafatar.com/renders/head/${UUIDInformation[playerName]}?overlay`}
          alt="Head not found"
        />),
      playerName: (
        <Typography.Text>{playerName}</Typography.Text>
      ),
    }
  ))
);

/**
 * About us page view.
 */
const AboutUsView = () => (
  <div className="about-us-content">
    <h1 className="about-us-title">Admins & Mods</h1>
    <List
      grid={{
        xxl: 3,
        xl: 3,
        lg: 3,
        md: 3,
        sm: 2,
        xs: 1,
      }}
      dataSource={buildHeadDisplayComponents(adminModsUUID)}
      renderItem={(component) => (
        <List.Item>
          <Card
            cover={component.image}
            bordered={false}
            size="small"
          >
            <Card.Meta
              className="about-us-card-subtitle"
              title={component.playerName}
            />
          </Card>
        </List.Item>
      )}
    />
    <h1 className="about-us-title">Veterans</h1>
    <List
      className="about-us-veterans-list"
      grid={{
        xxl: 5,
        xl: 5,
        lg: 5,
        md: 4,
        sm: 3,
        xs: 1,
      }}
      dataSource={veterans}
      renderItem={(player) => (
        <List.Item>
          <Typography.Text>
            {player}
          </Typography.Text>
        </List.Item>
      )}
    />
    <div className="about-us-crafatar">
      {'Thank you to '}
      <a href="https://crafatar.com">Crafatar</a>
      {' for providing avatars.'}
    </div>
  </div>
);

export default AboutUsView;
