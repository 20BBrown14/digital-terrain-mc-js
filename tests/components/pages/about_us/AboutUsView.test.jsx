import React from 'react';
import AboutUsView from '../../../../src/components/pages/about_us/AboutUsView';

describe('AboutUsView', () => {
  describe('view states', () => {
    it('renders a default view', () => {
      const view = render(
        <AboutUsView
          isLoading={false}
          hasServiceFailure={false}
          aboutUsInformation={{}}
          veteransInformation={[]}
        />,
      );
      expect(view).toMatchSnapshot();
    });

    it('renders a loading view', () => {
      const view = render(
        <AboutUsView
          isLoading
          hasServiceFailure={false}
          aboutUsInformation={{}}
          veteransInformation={[]}
        />,
      );

      expect(view).toMatchSnapshot();
    });

    it('renders an error view', () => {
      const view = render(
        <AboutUsView
          isLoading={false}
          hasServiceFailure
          aboutUsInformation={{}}
          veteransInformation={[]}
        />,
      );

      expect(view).toMatchSnapshot();
    });
  });
});
