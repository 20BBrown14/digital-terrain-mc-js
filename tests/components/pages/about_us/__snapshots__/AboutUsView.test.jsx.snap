// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsView renders a default view 1`] = `
<div
  class="about-us-content"
>
  <h1
    class="about-us-title"
  >
    Admins & Mods
  </h1>
  <div
    class="ant-list ant-list-split ant-list-grid"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-row"
        >
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <div
                  class="ant-card ant-card-small"
                >
                  <div
                    class="ant-card-cover"
                  >
                    <img
                      alt="Head not found"
                      class="about-us-card-cover"
                      src="https://crafatar.com/renders/head/4e9c8a2a-a9a4-4370-a4f7-1f56ffd65461?overlay"
                    />
                  </div>
                  <div
                    class="ant-card-body"
                  >
                    <div
                      class="ant-card-meta about-us-card-subtitle"
                    >
                      <div
                        class="ant-card-meta-detail"
                      >
                        <div
                          class="ant-card-meta-title"
                        >
                          <span
                            class="ant-typography"
                          >
                            Panda_Brah
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <div
                  class="ant-card ant-card-small"
                >
                  <div
                    class="ant-card-cover"
                  >
                    <img
                      alt="Head not found"
                      class="about-us-card-cover"
                      src="https://crafatar.com/renders/head/be65c2e6-7d20-47b6-9689-2da9584a7177?overlay"
                    />
                  </div>
                  <div
                    class="ant-card-body"
                  >
                    <div
                      class="ant-card-meta about-us-card-subtitle"
                    >
                      <div
                        class="ant-card-meta-detail"
                      >
                        <div
                          class="ant-card-meta-title"
                        >
                          <span
                            class="ant-typography"
                          >
                            DMDF_
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <div
                  class="ant-card ant-card-small"
                >
                  <div
                    class="ant-card-cover"
                  >
                    <img
                      alt="Head not found"
                      class="about-us-card-cover"
                      src="https://crafatar.com/renders/head/159a48ea-7a2f-4639-b176-f2b4f57af6d4?overlay"
                    />
                  </div>
                  <div
                    class="ant-card-body"
                  >
                    <div
                      class="ant-card-meta about-us-card-subtitle"
                    >
                      <div
                        class="ant-card-meta-detail"
                      >
                        <div
                          class="ant-card-meta-title"
                        >
                          <span
                            class="ant-typography"
                          >
                            Faeppuccino
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <div
                  class="ant-card ant-card-small"
                >
                  <div
                    class="ant-card-cover"
                  >
                    <img
                      alt="Head not found"
                      class="about-us-card-cover"
                      src="https://crafatar.com/renders/head/027d8c7c-509c-466a-93a2-966a2af78f1a?overlay"
                    />
                  </div>
                  <div
                    class="ant-card-body"
                  >
                    <div
                      class="ant-card-meta about-us-card-subtitle"
                    >
                      <div
                        class="ant-card-meta-detail"
                      >
                        <div
                          class="ant-card-meta-title"
                        >
                          <span
                            class="ant-typography"
                          >
                            DadsaMuggle
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <div
                  class="ant-card ant-card-small"
                >
                  <div
                    class="ant-card-cover"
                  >
                    <img
                      alt="Head not found"
                      class="about-us-card-cover"
                      src="https://crafatar.com/renders/head/94a16372-984d-463d-b416-60acd455cc75?overlay"
                    />
                  </div>
                  <div
                    class="ant-card-body"
                  >
                    <div
                      class="ant-card-meta about-us-card-subtitle"
                    >
                      <div
                        class="ant-card-meta-detail"
                      >
                        <div
                          class="ant-card-meta-title"
                        >
                          <span
                            class="ant-typography"
                          >
                            Kattatonic
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h1
    class="about-us-title"
  >
    Veterans
  </h1>
  <div
    class="ant-list about-us-veterans-list ant-list-split ant-list-grid"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-row"
        >
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Animvz
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Barbosa1
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  CallMeAces
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Deynum
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  FalconRx
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Jupiterbeans
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Llarry_Llama
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Malsiv
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Sasefelt
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  SaytotheRah
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  TheAltius
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  TheRealCutlass
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  yGlobalista
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="flex:1 1 auto"
            >
              <div
                class="ant-list-item"
              >
                <span
                  class="ant-typography"
                >
                  Yoshkeen
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="about-us-crafatar"
  >
    Thank you to 
    <a
      href="https://crafatar.com"
    >
      Crafatar
    </a>
     for providing avatars.
  </div>
</div>
`;
